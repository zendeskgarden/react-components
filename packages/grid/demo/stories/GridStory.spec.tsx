/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { render } from 'garden-test-utils';
import 'jest-styled-components';
import { GridStory } from './GridStory';

const renderAndMatchSnapshot = (props: any) => {
  const { container } = render(<GridStory {...props} />);
  expect(container.firstChild).toMatchSnapshot();
};

describe('GridStory Component', () => {
  it('renders default GridStory', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ]
    });
  });

  it('renders GridStory with alignItems', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      alignItems: 'center' // Must be one of ['start', 'end', 'center', 'baseline', 'stretch']
    });
  });

  it('renders GridStory with alignItems for different screen sizes', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      alignItemsXs: 'start', // Must be one of ['start', 'end', 'center', 'baseline', 'stretch']
      alignItemsSm: 'center',
      alignItemsMd: 'end',
      alignItemsLg: 'baseline',
      alignItemsXl: 'stretch'
    });
  });

  it('renders GridStory with justifyContent', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      justifyContent: 'between' // Must be one of ['start', 'end', 'center', 'between', 'around']
    });
  });

  it('renders GridStory with justifyContent for different screen sizes', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      justifyContentXs: 'start', // Must be one of ['start', 'end', 'center', 'between', 'around']
      justifyContentSm: 'center',
      justifyContentMd: 'end',
      justifyContentLg: 'around',
      justifyContentXl: 'between'
    });
  });

  it('renders GridStory with wrap', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      wrap: 'wrap' // Must be one of ['nowrap', 'wrap', 'wrap-reverse']
    });
  });

  it('renders GridStory with wrap for different screen sizes', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      wrapXs: 'nowrap', // Must be one of ['nowrap', 'wrap', 'wrap-reverse']
      wrapSm: 'wrap',
      wrapMd: 'wrap-reverse',
      wrapLg: 'wrap',
      wrapXl: 'nowrap'
    });
  });

  it('renders GridStory with multiple rows and columns', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 4 }, { size: 4 }, { size: 4 }]
        },
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ]
    });
  });

  it('renders GridStory with alignItems, justifyContent, and wrap combined', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      alignItems: 'center', // Must be one of ['start', 'end', 'center', 'baseline', 'stretch']
      justifyContent: 'between', // Must be one of ['start', 'end', 'center', 'between', 'around']
      wrap: 'wrap' // Must be one of ['nowrap', 'wrap', 'wrap-reverse']
    });
  });

  it('renders GridStory with alignItems, justifyContent, and wrap for different screen sizes', () => {
    renderAndMatchSnapshot({
      rows: [
        {
          cols: [{ size: 6 }, { size: 6 }]
        }
      ],
      alignItemsXs: 'start', // Must be one of ['start', 'end', 'center', 'baseline', 'stretch']
      alignItemsSm: 'center',
      alignItemsMd: 'end',
      alignItemsLg: 'baseline',
      alignItemsXl: 'stretch',
      justifyContentXs: 'start', // Must be one of ['start', 'end', 'center', 'between', 'around']
      justifyContentSm: 'center',
      justifyContentMd: 'end',
      justifyContentLg: 'around',
      justifyContentXl: 'between',
      wrapXs: 'nowrap', // Must be one of ['nowrap', 'wrap', 'wrap-reverse']
      wrapSm: 'wrap',
      wrapMd: 'wrap-reverse',
      wrapLg: 'wrap',
      wrapXl: 'nowrap'
    });
  });
});
