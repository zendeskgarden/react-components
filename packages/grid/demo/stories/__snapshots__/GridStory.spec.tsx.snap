// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridStory Component renders GridStory with alignItems 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with alignItems for different screen sizes 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {
  .c1 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (min-width:576px) {
  .c1 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

@media (min-width:992px) {
  .c1 {
    -webkit-align-items: baseline;
    -webkit-box-align: baseline;
    -ms-flex-align: baseline;
    align-items: baseline;
  }
}

@media (min-width:1200px) {
  .c1 {
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with alignItems, justifyContent, and wrap combined 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with alignItems, justifyContent, and wrap for different screen sizes 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:576px) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-wrap: wrap-reverse;
    -ms-flex-wrap: wrap-reverse;
    flex-wrap: wrap-reverse;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (min-width:992px) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-align-items: baseline;
    -webkit-box-align: baseline;
    -ms-flex-align: baseline;
    align-items: baseline;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (min-width:1200px) {
  .c1 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with justifyContent 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with justifyContent for different screen sizes 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {
  .c1 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:576px) {
  .c1 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (min-width:992px) {
  .c1 {
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (min-width:1200px) {
  .c1 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with multiple rows and columns 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 33.33333333333333%;
  -ms-flex-preferred-size: 33.33333333333333%;
  flex-basis: 33.33333333333333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 33.33333333333333%;
  padding-right: 10px;
  padding-left: 10px;
}

.c3 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c3"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c3"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with wrap 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders GridStory with wrap for different screen sizes 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

@media (min-width:576px) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-wrap: wrap-reverse;
    -ms-flex-wrap: wrap-reverse;
    flex-wrap: wrap-reverse;
  }
}

@media (min-width:992px) {
  .c1 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

@media (min-width:1200px) {
  .c1 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;

exports[`GridStory Component renders default GridStory 1`] = `
.c2 {
  box-sizing: border-box;
  width: 100%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  max-width: 50%;
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  box-sizing: border-box;
  padding-right: 10px;
  padding-left: 10px;
  color-scheme: only light;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -10px;
  margin-left: -10px;
}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (min-width:0px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="c0"
  data-garden-id="grid.grid"
  data-garden-version="9.1.1"
>
  <div
    class="c1"
    data-garden-id="grid.row"
    data-garden-version="9.1.1"
  >
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
    <div
      class="c2"
      data-garden-id="grid.col"
      data-garden-version="9.1.1"
    />
  </div>
</div>
`;
